### PR TITLE
Refactor access-code component test file to new format

### PR DESCRIPTION
**Pull request by balibirchlee**
### What is the context of this PR?

Fixes [#200](https://github.com/ONSdigital/design-team/issues/200) in the design team issues board

This PR refactors the macrospec testing file for the access-code component to follow the new testing format.

### How to review this PR

Ask yourself....
- Do we need any additional tests?
- Are the tests descriptions clear?
- Does the order of the test document make sense?
- Is it internally consistent?
- Is it consistent with the [guidelines for test refactoring](https://confluence.ons.gov.uk/display/SDC/Testing+Refactor+Guidance)? 

run this test command to view the test results:
`yarn test --testNamePattern="FOR: access-code"`

### Checklist

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/access-code/_macro.spec.js [hunk 1]
```diff
@@ -5,150 +5,184 @@ import * as cheerio from 'cheerio';
...
+            const $ = cheerio.load(
+                renderComponent('access-code', {
+                    groupSize: 2,
+                }),
+            );
+            test('THEN: renders input with provided groupsize', () => {
+                expect($('input').attr('data-group-size')).toBe('2');
+            });
+        });
+        describe('WHEN: maxlength and groupsize params are provided', () => {
```
Comment (rmccar) [inline src/components/access-code/_macro.spec.js:135]:
```suggestion
        describe('WHEN: maxlength and groupSize params are provided', () => {
```

File: src/components/access-code/_macro.spec.js [hunk 2]
```diff
@@ -5,150 +5,184 @@ import * as cheerio from 'cheerio';
...
+            const $ = cheerio.load(
+                renderComponent('access-code', {
+                    maxlength: 8,
+                }),
+            );
+            test('THEN: renders input with provided maxlength', () => {
+                expect($('input').attr('maxlength')).toBe('9');
+            });
+        });
+        describe('WHEN: groupsize param is provided', () => {
```
Comment (rmccar) [inline src/components/access-code/_macro.spec.js:125]:
```suggestion
        describe('WHEN: groupSize param is provided', () => {
```

File: src/components/access-code/_macro.spec.js [hunk 3]
```diff
@@ -5,150 +5,184 @@ import * as cheerio from 'cheerio';
...
+            });
+        });
+        describe('WHEN: maxlength and groupsize params are provided', () => {
+            const $ = cheerio.load(
+                renderComponent('access-code', {
+                    maxlength: 6,
+                    groupSize: 3,
+                }),
+            );
+            test('THEN: renders input with provided maxlength accounting for provided groupsize', () => {
```
Comment (rmccar) [inline src/components/access-code/_macro.spec.js:142]:
```suggestion
            test('THEN: renders input with provided maxlength accounting for provided groupSize', () => {
```

File: src/components/access-code/_macro.spec.js [hunk 4]
```diff
@@ -5,150 +5,184 @@ import * as cheerio from 'cheerio';
...
-        expect($('.ons-input--with-description').text()).toBe(
-            'Keep this code safe. You will need to enter it every time you access your study',
-        );
+    describe('GIVEN: Params: maxlength & groupSize', () => {
+        describe('WHEN: params are at default state', () => {
+            const $ = cheerio.load(renderComponent('access-code'));
+            test('THEN: renders input with total maxlength of 19', () => {
+                expect($('input').attr('maxlength')).toBe('19');
+            });
+            test('THEN: renders input with groupsize of 4', () => {
```
Comment (rmccar) [inline src/components/access-code/_macro.spec.js:111]:
```suggestion
            test('THEN: renders input with groupSize of 4', () => {
```

**Review by rmccar — COMMENTED** [review]
Other than these minor things, looks good

**Review by alessioventuriniAND — APPROVED** [review]

**Pull request merged**